pandas/core/window/common.py

```diff
@@ -83,8 +83,18 @@ def dataframe_from_int_dict(data, frame_template):
                         # mypy needs to know columns is a MultiIndex, Index doesn't
                         # have levels attribute
                         arg2.columns = cast(MultiIndex, arg2.columns)
-                        result.index = MultiIndex.from_product(
-                            arg2.columns.levels + [result_index]
+                        # GH 21157
```

Comment:
thi s is really confusing. you are matching the codes from the original? can you reindex?

Reply:
I'll add comment to what's going on here, but unfortunately can't reindex here.

This needs to act like a `MultiIndex.product`, but instead of taking the product of all original levels + result level independently, the original levels need too be treated like a group first before taking a product